### PR TITLE
Issue #3275: Global Operator State

### LINKED ISSUE #3275: row_number() window function is broken on views since 0.3.2
#### What happens?
The `row_number()` window function worked on views in 0.3.1 but is broken since 0.3.2

#### To Reproduce

```sql
DROP TABLE IF EXISTS v1;
DROP TABLE IF EXISTS v2;
DROP VIEW IF EXISTS vertices_view;

CREATE TABLE v1(id bigint);
CREATE TABLE v2(id bigint);

INSERT INTO v1 VALUES (1),  (2),  (3);
INSERT INTO v2 VALUES (1),  (2);

CREATE VIEW vertices_view AS
  SELECT * FROM v1
  UNION ALL
  SELECT * FROM v2;

SELECT id AS sparse_id, row_number() OVER () AS rnum FROM vertices_view;
```

On 0.3.1, it gives the following result:
```
┌───────────┬──────┐
│ sparse_id │ rnum │
├───────────┼──────┤
│ 1         │ 1    │
│ 2         │ 2    │
│ 1         │ 3    │
│ 2         │ 4    │
│ 3         │ 5    │
└───────────┴──────┘
```

On 0.3.2+, it gives:
```
┌───────────┬──────┐
│ sparse_id │ rnum │
├───────────┼──────┤
│ 1         │ 1    │
│ 2         │ 2    │
│ 3         │ 3    │
│ 1         │ 1    │
│ 2         │ 2    │
└───────────┴──────┘
```

The first result is the correct one.

If the view is materialized to a table, both versions return the correct results.

```
DROP TABLE IF EXISTS vertices_table;
CREATE TABLE vertices_table(id bigint);
INSERT INTO vertices_table
  SELECT * FROM v1
  UNION ALL
  SELECT * FROM v2;
SELECT id AS sparse_id, row_number() OVER () AS rnum FROM vertices_table;
```
```
┌───────────┬──────┐
│ sparse_id │ rnum │
├───────────┼──────┤
│ 1         │ 1    │
│ 2         │ 2    │
│ 3         │ 3    │
│ 1         │ 4    │
│ 2         │ 5    │
└───────────┴──────┘
```

#### Environment (please complete the following information):
 - OS: Fedora 35
 - DuckDB Version: 0.3.2+
 - DuckDB Client: CLI

### PR DESCRIPTION
**Pull request by hawkfish**
Add a global operator state for streaming operators.
This ensures that streaming window does not reset the row number
when run by multiple pipelines.

### CODE REVIEW

File: test/sql/window/test_streaming_window.test [hunk 1]
```diff
@@ -131,3 +131,33 @@ query TT
 explain select last_value(i) over (order by j), first_value(i) over () from integers
 ----
 physical_plan	<REGEX>:.*STREAMING_WINDOW.*WINDOW.*
+
+
+statement ok
+CREATE TABLE v1(id bigint);
```
Comment (Mytherin) [inline test/sql/window/test_streaming_window.test:137]: Could we add some tests where we use the streaming window operator in:

* Recursive CTEs
* In a prepared statement that is executed multiple times

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good to me. One request for some more tests:

**Comment by Mytherin** [comment]
The CI error here is related to an invisible merge conflict that I fixed in #3291.

**Comment by Mytherin** [comment]
Could you merge with master? The CI error should be fixed now.

**Review by lnkuiper — APPROVED** [review]

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**